src/main/scala/it/pagopa/interop/authorizationprocess/service/impl/AuthorizationManagementServiceImpl.scala

```diff
@@ -168,7 +168,11 @@ final case class AuthorizationManagementServiceImpl(
       keyApi.createKeys(xCorrelationId = correlationId, clientId, keysSeeds, xForwardedFor = ip)(
         BearerToken(bearerToken)
       )
-    invoker.invoke(request, "Key creation")
+    invoker
+      .invoke(request, "Key creation")
+      .recoverWith {
+        case err: ApiError[_] if err.code == 404 => Future.failed(ClientNotFound(clientId))
```

Comment:
I see why this has been added, but I think the issue is shadier than this 🥷 
Before calling the `createKeys`, a `getClient` is executed, and if the client was actually missing, an error should have been raised before arriving here.

I'm ok if it breaks here because it means that something is very wrong somewhere (like it seems to be 🤔 )

If this makes sense to you too, I would remove this part and open a different bug to investigate

Reply:
Fixed in the last commit